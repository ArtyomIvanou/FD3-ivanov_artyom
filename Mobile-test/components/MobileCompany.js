import React from 'react';
import PropTypes from 'prop-types';

import MobileClient from './MobileClient';
import MobileClientEditor from './MobileClientEditor';
import './MobileCompany.css';

import { editEvents } from './events';

class MobileCompany extends React.PureComponent {

  static propTypes = {
    name: PropTypes.string.isRequired,
    clients: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.number.isRequired,
        fio: PropTypes.string.isRequired,
        balance: PropTypes.number.isRequired,
      })
    ),
  };

  state = {
    name: this.props.name,
    clients: this.props.clients,
    edited: false,
    editNum: null,
    created:false,
  };
  componentDidMount = () => {
    editEvents.addListener('DeleteClient', this.deteteClient);
    editEvents.addListener('EditClient', this.editClient);
    editEvents.addListener('SaveClient', this.saveClient);
    editEvents.addListener('SaveNewClient', this.saveNewClient);
   
  };

  componentWillUnmount = () => {
    editEvents.removeListener('DeleteClient', this.deteteClient);
    editEvents.removeListener('EditClient', this.editClient);
    editEvents.removeListener('SaveClient', this.saveClient);
    editEvents.removeListener('SaveNewClient', this.saveNewClient);
   
  };
  allClients = this.props.clients
  hashOfNewClient={id:null, fio:null,fam:null, im:null, otch:null,status:null, balance:null}
  qqqq = (mode) => {
    if (mode == 1) {//все клиенты

      var ccc = [...this.allClients]; // копия самого массива клиентов

    }
    if (mode == 2) {//активные клиенты

      let newClients = [...this.allClients]; // копия самого массива клиентов

      var ccc = newClients.filter(function (s, i) {

        return s.status != "blocked"
      })

    }
    if (mode == 3) {//заблокированные клиенты
      let newClients = [...this.allClients]; // копия самого массива клиентов

      var ccc = newClients.filter(function (s, i) {

        return s.status != "active"
      })

    }
    this.setState({ clients: ccc });

  };
  showAll = () => {
    this.qqqq(1)
  };
  showSortActive = () => {
    this.qqqq(2)

  };
  showSortBlocked = () => {
    this.qqqq(3)

  };

  deteteClient = (code) => {
    let newClients = [...this.state.clients]; // копия самого массива клиентов
    let ccc = newClients.filter(function (s, i) {

      return s.id != code
    })
    this.allClients = this.allClients.filter(function (s, i) {

      return s.id != code
    })
    this.setState({ clients: ccc });
  };
  editClient = (code) => {
    this.setState({ created: false });
    this.setState({ edited: true })
    var ttt = null
    let newClients = [...this.state.clients]; // копия самого массива клиентов
    let ccc = newClients.filter(function (s, i) {
      if (s.id == code) {
        ttt = i
        return
      } 
    })
    console.log(ttt)
    this.setState({ editNum: ttt })
  };
  saveClient = (code,hash) => {
    this.setState({ created: false });
    this.setState({ edited: false })
    this.setState({ editNum: null })
    var ttt = null
    var aaa=null
    let newClients = [...this.state.clients]; // копия самого массива клиентов
    let ccc = newClients.filter(function (s, i) {
      if (s.id == code) {
        ttt = i
      } 
    })
    let www=this.allClients.filter(function (s, i) {
      if (s.id == code) {
        aaa = i
      } 
    })
    this.allClients[aaa]=hash
    newClients[ttt]=hash
    this.setState({ clients: newClients})
    
  };
  saveNewClient = (hash) => {
    
    this.setState({ created: false });
    this.setState({ edited: false })
    this.setState({ editNum: null })
    let newClients = [...this.state.clients]// копия самого массива клиентов
    newClients.push(hash)
    this.allClients.push(hash)
    this.setState({ clients: newClients})
    
  };
  addNewClient = () => {
    this.setState({ edited: false });
    this.setState({ created: true });
  };
  setName1 = () => {
    this.setState({ name: 'МТС' });
  };

  setName2 = () => {
    this.setState({ name: 'Velcom' });
  };




 

  render() {
    
    console.log("MobileCompany render");
    if (this.state.edited) {

      var mobileClientEditor= <MobileClientEditor key={this.allClients[this.state.editNum].id} info={this.allClients[this.state.editNum]} />
    }
    else{var mobileClientEditor=null}
    if (this.state.created) {
      var mobileClientCreator= <MobileClientEditor key={this.allClients.length+1} info={this.hashOfNewClient} newClient={true}/>
    } else {
      var mobileClientCreator=null
    }
    var clientsCode = this.state.clients.map(client =>
      <MobileClient key={client.id} info={client} />
    );
    return (
      <div className='MobileCompany'>
        <input type="button" value="Velcom" onClick={this.setName2} />
        <input type="button" value="МТС" onClick={this.setName1} />

        <div className='MobileCompanyName'>Компания: {this.state.name}</div>

        <div className="CliensSorting">
          <input type="button" value="Все" onClick={this.showAll} />
          <input type="button" value="Активные" onClick={this.showSortActive} />
          <input type="button" value="Заблокированные" onClick={this.showSortBlocked} />
        </div>
        <table className='MobileCompanyClients'>
          <tbody className='Answers' >
            <tr><td>фамилия</td><td>имя</td><td>отчество</td><td>статус</td><td>баланс</td><td>редактировать</td><td>удалить</td></tr>
          </tbody>
          <tbody className='Answers' >{clientsCode}</tbody>
        </table>
       
        <div><input type="button" value="Добавить нового клиента" onClick={this.addNewClient}/></div>
        <div>{mobileClientEditor}</div>
        <div>{mobileClientCreator}</div>
      </div>
    )
    
      ;

  }

}

export default MobileCompany;
