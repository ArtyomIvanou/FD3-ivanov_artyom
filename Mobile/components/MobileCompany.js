import React from 'react';
import PropTypes from 'prop-types';

import MobileClient from './MobileClient';

import './MobileCompany.css';
import { editEvents } from './events';
class MobileCompany extends React.PureComponent {

  static propTypes = {
    name: PropTypes.string.isRequired,
    clients: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.number.isRequired,
        fio: PropTypes.string.isRequired,
        balance: PropTypes.number.isRequired,
      })
    ),
  };

  state = {
    name: this.props.name,
    clients: this.props.clients,
  };
  componentDidMount = () => {
    editEvents.addListener('DeleteClient', this.deteteClient);
    //editEvents.addListener('EFreeAnswerTextChanged',this.freeAnswerTextChanged);
  };

  componentWillUnmount = () => {
    editEvents.removeListener('DeleteClient', this.deteteClient);
    //editEvents.removeListener('EFreeAnswerTextChanged',this.freeAnswerTextChanged);
  };
  showAll = () => {
    let newClients = [...this.state.clients]; // копия самого массива клиентов
    
      
    
    this.setState({ clients:newClients });
  };
  showSort = () => {
    let newClients = [...this.state.clients]; // копия самого массива клиентов
    
      let ccc = newClients.filter(function (s, i) {

        return s.status != "blocked"
      })
    
    this.setState({ clients: ccc });
  };
  deteteClient = (code) => {
    let newClients = [...this.state.clients]; // копия самого массива клиентов
    let ccc = newClients.filter(function (s, i) {

      return s.id != code
    })
    
    this.setState({ clients: ccc });
  };
  setName1 = () => {
    this.setState({ name: 'МТС' });
  };

  setName2 = () => {
    this.setState({ name: 'Velcom' });
  };




  setBalance = (clientId, newBalance) => {
    let changed = false;
    let newClients = [...this.state.clients]; // копия самого массива клиентов
    newClients.forEach((c, i) => {
      if (c.id == clientId && c.balance != newBalance) {
        let newClient = { ...c }; // копия хэша изменившегося клиента
        newClient.balance = newBalance;
        newClients[i] = newClient;
        changed = true;
      }
    });
    if (changed)
      this.setState({ clients: newClients });
  };


  setBalance1 = () => {
    this.setBalance(375292223345, 230);
  };

  setBalance2 = () => {
    this.setBalance(375292223345, 250);
  };

  render() {

    console.log("MobileCompany render");

    var clientsCode = this.state.clients.map(client =>
      <MobileClient key={client.id} info={client} />
    );
    return (
      <div className='MobileCompany'>
        <input type="button" value="Velcom" onClick={this.setName2} />
        <input type="button" value="МТС" onClick={this.setName1} />

        <div className='MobileCompanyName'>Компания: {this.state.name}</div>

        <div className="CliensSorting">
          <input type="button" value="Все" onClick={this.showAll} />
          <input type="button" value="Активные" onClick={this.showSort} />
          <input type="button" value="Заблокированные" onClick={this.showSort}/>
        </div>
        <table className='MobileCompanyClients'>
          <tbody className='Answers' >
            <tr><td>имя</td><td>фамилия</td><td>отчество</td><td>статус</td><td>баланс</td><td>редактировать</td><td>удалить</td></tr>
          </tbody>
          <tbody className='Answers' >{clientsCode}</tbody>
        </table>
        <div> <input type="button" value="Сидоров=230" onClick={this.setBalance1} />
          <input type="button" value="Сидоров=250" onClick={this.setBalance2} /></div>
        <div><input type="button" value="Добавить нового клиента" /></div>
      </div>
    )
      /*return (
        <div className='MobileCompany'>
          <input type="button" value="Velcom" onClick={this.setName2} />
          <input type="button" value="МТС" onClick={this.setName1} />
         
          <div className='MobileCompanyName'>Компания: {this.state.name}</div>
  
          <div className="CliensSorting">
          <input type="button" value="Все"  />
          <input type="button" value="Активные"  />
          <input type="button" value="Заблокированные" />
          </div>
  
          <div className='MobileCompanyClients'>
            {clientsCode}
          </div>
          <input type="button" value="Сидоров=230" onClick={this.setBalance1} />
          <input type="button" value="Сидоров=250" onClick={this.setBalance2} />
        </div>
      )*/
      ;

  }

}

export default MobileCompany;
