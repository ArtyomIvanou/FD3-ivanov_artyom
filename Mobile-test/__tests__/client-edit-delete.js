import React from 'react';
import renderer from 'react-test-renderer';

import MobileCompany from '../components/MobileCompany';

test('работа кнопки edit and delete', () => {
  //массив клиентов
  let clientsArr = [
    { id: 375292223344, fio: "Иванов И.И.", fam: "Иванов", im: "Иван", otch: "Иванович", status: "active", balance: 200 },
    { id: 375292223345, fio: "Сидоров С.С.", fam: "Сидоров", im: "Сидор", otch: "Сидорович", status: "active", balance: 250 },
    { id: 375292223346, fio: "Петров П.П.", fam: "Петров", im: "Пётр", otch: "Петрович", status: "active", balance: 180 },
    { id: 375292223347, fio: "Григорьев Г.Г.", fam: "Григорьев", im: "Григорий", otch: "Григорьевич", status: "blocked", balance: -220 },
  ];
//создаем компонент
  const component = renderer.create(
    <MobileCompany clients={clientsArr} key={'1'} />
  );
  let componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();

//ищем клиента для редактирования
  const buttonElemNew = component.root.find(el => el.type == 'input' && el.props.id == '375292223344Edit');
  
  
  buttonElemNew.props.onClick();
  componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();
  
  //новые данные клиента
  var newClienta = { id: 375292223344, fio: "Александров А.А", fam: "Александров", im: "Александр", otch: "Александрович", status: "active", balance: 500 }



  const inputNewNumb = component.root.find(el => el.type == 'input' && el.props.id == "newNumber");
 //печатаем
  inputNewNumb.ref = newClienta.id;
 //печатаем
  const inputNewFam = component.root.find(el => el.type == 'input' && el.props.id == "newFam");
  //печатаем
  inputNewFam.ref = newClienta.fam;
  const inputNewIm = component.root.find(el => el.type == 'input' && el.props.id == "newIm");
//печатаем
  inputNewIm.ref= newClienta.im;
  const inputNewOtch = component.root.find(el => el.type == 'input' && el.props.id == "newOtch");
 //выбираем
  inputNewOtch.ref= newClienta.otch;
  const inputNewStatus = component.root.find( el => el.type=='select' && el.props.id == "newStatus" ); 
 //печатаем
  inputNewStatus.ref=newClienta.status;
  const inputNewBalance = component.root.find(el => el.type == 'input' && el.props.id == "newBalance");
 //печатаем
  inputNewBalance.ref= newClienta.balance;



//сохраняем
  const buttonElemSaveNew = component.root.find(el => el.type == 'input' && el.props.id=='saveNewClient');
  // и "нажмём" на неё
  buttonElemSaveNew.props.onClick();


  componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();


//удаляем
  const buttonDelete = component.root.find(el => el.type == 'input' && el.props.id == '375292223347Delete');
  // и "нажмём" на неё
  //buttonElemNew.props.info=
  buttonDelete.props.onClick()
  componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();
});
