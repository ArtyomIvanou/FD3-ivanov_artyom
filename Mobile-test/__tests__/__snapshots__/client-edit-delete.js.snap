// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`работа кнопки edit and delete 1`] = `
<div
  className="MobileCompany"
>
  <div
    className="CliensSorting"
  >
    <input
      onClick={[Function]}
      type="button"
      value="Все"
    />
    <input
      onClick={[Function]}
      type="button"
      value="Активные"
    />
    <input
      onClick={[Function]}
      type="button"
      value="Заблокированные"
    />
  </div>
  <table
    className="MobileCompanyClients"
  >
    <tbody
      className="Answers"
    >
      <tr>
        <td>
          фамилия
        </td>
        <td>
          имя
        </td>
        <td>
          отчество
        </td>
        <td>
          статус
        </td>
        <td>
          баланс
        </td>
        <td>
          редактировать
        </td>
        <td>
          удалить
        </td>
      </tr>
    </tbody>
    <tbody
      className="Answers"
    >
      <tr
        className="MobileClient"
      >
        <td
          className="MobileClientFam"
        >
          Иванов
        </td>
        <td
          className="MobileClientIm"
        >
          Иван
        </td>
        <td
          className="MobileClientOtch"
        >
          Иванович
        </td>
        <td
          className="MobileClientStatus"
        >
          active
        </td>
        <td
          className="MobileClientBalance"
        >
          200
        </td>
        <td
          className="ButtonEdit"
        >
           
          <input
            id="375292223344Edit"
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td
          className="ButtonDelete"
        >
           
          <input
            id="375292223344Delete"
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr
        className="MobileClient"
      >
        <td
          className="MobileClientFam"
        >
          Сидоров
        </td>
        <td
          className="MobileClientIm"
        >
          Сидор
        </td>
        <td
          className="MobileClientOtch"
        >
          Сидорович
        </td>
        <td
          className="MobileClientStatus"
        >
          active
        </td>
        <td
          className="MobileClientBalance"
        >
          250
        </td>
        <td
          className="ButtonEdit"
        >
           
          <input
            id="375292223345Edit"
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td
          className="ButtonDelete"
        >
           
          <input
            id="375292223345Delete"
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr
        className="MobileClient"
      >
        <td
          className="MobileClientFam"
        >
          Петров
        </td>
        <td
          className="MobileClientIm"
        >
          Пётр
        </td>
        <td
          className="MobileClientOtch"
        >
          Петрович
        </td>
        <td
          className="MobileClientStatus"
        >
          active
        </td>
        <td
          className="MobileClientBalance"
        >
          180
        </td>
        <td
          className="ButtonEdit"
        >
           
          <input
            id="375292223346Edit"
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td
          className="ButtonDelete"
        >
           
          <input
            id="375292223346Delete"
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr
        className="MobileClient"
      >
        <td
          className="MobileClientFam"
        >
          Григорьев
        </td>
        <td
          className="MobileClientIm"
        >
          Григорий
        </td>
        <td
          className="MobileClientOtch"
        >
          Григорьевич
        </td>
        <td
          className="MobileClientStatus"
        >
          blocked
        </td>
        <td
          className="MobileClientBalance"
        >
          -220
        </td>
        <td
          className="ButtonEdit"
        >
           
          <input
            id="375292223347Edit"
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td
          className="ButtonDelete"
        >
           
          <input
            id="375292223347Delete"
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
    </tbody>
  </table>
  <div>
    <input
      onClick={[Function]}
      type="button"
      value="Добавить нового клиента"
    />
  </div>
  <div />
  <div />
</div>
`;

exports[`работа кнопки edit and delete 2`] = `
<div
  className="MobileCompany"
>
  <div
    className="CliensSorting"
  >
    <input
      onClick={[Function]}
      type="button"
      value="Все"
    />
    <input
      onClick={[Function]}
      type="button"
      value="Активные"
    />
    <input
      onClick={[Function]}
      type="button"
      value="Заблокированные"
    />
  </div>
  <table
    className="MobileCompanyClients"
  >
    <tbody
      className="Answers"
    >
      <tr>
        <td>
          фамилия
        </td>
        <td>
          имя
        </td>
        <td>
          отчество
        </td>
        <td>
          статус
        </td>
        <td>
          баланс
        </td>
        <td>
          редактировать
        </td>
        <td>
          удалить
        </td>
      </tr>
    </tbody>
    <tbody
      className="Answers"
    >
      <tr
        className="MobileClient"
      >
        <td
          className="MobileClientFam"
        >
          Иванов
        </td>
        <td
          className="MobileClientIm"
        >
          Иван
        </td>
        <td
          className="MobileClientOtch"
        >
          Иванович
        </td>
        <td
          className="MobileClientStatus"
        >
          active
        </td>
        <td
          className="MobileClientBalance"
        >
          200
        </td>
        <td
          className="ButtonEdit"
        >
           
          <input
            id="375292223344Edit"
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td
          className="ButtonDelete"
        >
           
          <input
            id="375292223344Delete"
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr
        className="MobileClient"
      >
        <td
          className="MobileClientFam"
        >
          Сидоров
        </td>
        <td
          className="MobileClientIm"
        >
          Сидор
        </td>
        <td
          className="MobileClientOtch"
        >
          Сидорович
        </td>
        <td
          className="MobileClientStatus"
        >
          active
        </td>
        <td
          className="MobileClientBalance"
        >
          250
        </td>
        <td
          className="ButtonEdit"
        >
           
          <input
            id="375292223345Edit"
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td
          className="ButtonDelete"
        >
           
          <input
            id="375292223345Delete"
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr
        className="MobileClient"
      >
        <td
          className="MobileClientFam"
        >
          Петров
        </td>
        <td
          className="MobileClientIm"
        >
          Пётр
        </td>
        <td
          className="MobileClientOtch"
        >
          Петрович
        </td>
        <td
          className="MobileClientStatus"
        >
          active
        </td>
        <td
          className="MobileClientBalance"
        >
          180
        </td>
        <td
          className="ButtonEdit"
        >
           
          <input
            id="375292223346Edit"
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td
          className="ButtonDelete"
        >
           
          <input
            id="375292223346Delete"
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr
        className="MobileClient"
      >
        <td
          className="MobileClientFam"
        >
          Григорьев
        </td>
        <td
          className="MobileClientIm"
        >
          Григорий
        </td>
        <td
          className="MobileClientOtch"
        >
          Григорьевич
        </td>
        <td
          className="MobileClientStatus"
        >
          blocked
        </td>
        <td
          className="MobileClientBalance"
        >
          -220
        </td>
        <td
          className="ButtonEdit"
        >
           
          <input
            id="375292223347Edit"
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td
          className="ButtonDelete"
        >
           
          <input
            id="375292223347Delete"
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
    </tbody>
  </table>
  <div>
    <input
      onClick={[Function]}
      type="button"
      value="Добавить нового клиента"
    />
  </div>
  <div>
    <div
      className="MobileClientEdit"
    >
      <div>
        Номер
        <input
          defaultValue={375292223344}
          id="newNumber"
          type="text"
        />
      </div>
      <div>
        Фамилия
        <input
          defaultValue="Иванов"
          id="newFam"
          type="text"
        />
      </div>
      <div>
        Имя
        <input
          defaultValue="Иван"
          id="newIm"
          type="text"
        />
      </div>
      <div>
        Отчество
        <input
          defaultValue="Иванович"
          id="newOtch"
          type="text"
        />
      </div>
      <div>
        Статус
        <select
          defaultValue="blocked"
          id="newStatus"
        >
          <option
            value="active"
          >
            active
          </option>
          <option
            value="blocked"
          >
            blocked
          </option>
        </select>
      </div>
      <div>
        Баланс
        <input
          defaultValue={200}
          id="newBalance"
          type="text"
        />
      </div>
      <input
        id="saveNewClient"
        onClick={[Function]}
        type="button"
        value="Сохранить"
      />
    </div>
  </div>
  <div />
</div>
`;

exports[`работа кнопки edit and delete 3`] = `
<div
  className="MobileCompany"
>
  <div
    className="CliensSorting"
  >
    <input
      onClick={[Function]}
      type="button"
      value="Все"
    />
    <input
      onClick={[Function]}
      type="button"
      value="Активные"
    />
    <input
      onClick={[Function]}
      type="button"
      value="Заблокированные"
    />
  </div>
  <table
    className="MobileCompanyClients"
  >
    <tbody
      className="Answers"
    >
      <tr>
        <td>
          фамилия
        </td>
        <td>
          имя
        </td>
        <td>
          отчество
        </td>
        <td>
          статус
        </td>
        <td>
          баланс
        </td>
        <td>
          редактировать
        </td>
        <td>
          удалить
        </td>
      </tr>
    </tbody>
    <tbody
      className="Answers"
    >
      <tr
        className="MobileClient"
      >
        <td
          className="MobileClientFam"
        />
        <td
          className="MobileClientIm"
        />
        <td
          className="MobileClientOtch"
        />
        <td
          className="MobileClientStatus"
        />
        <td
          className="MobileClientBalance"
        />
        <td
          className="ButtonEdit"
        >
           
          <input
            id="undefinedEdit"
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td
          className="ButtonDelete"
        >
           
          <input
            id="undefinedDelete"
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr
        className="MobileClient"
      >
        <td
          className="MobileClientFam"
        >
          Сидоров
        </td>
        <td
          className="MobileClientIm"
        >
          Сидор
        </td>
        <td
          className="MobileClientOtch"
        >
          Сидорович
        </td>
        <td
          className="MobileClientStatus"
        >
          active
        </td>
        <td
          className="MobileClientBalance"
        >
          250
        </td>
        <td
          className="ButtonEdit"
        >
           
          <input
            id="375292223345Edit"
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td
          className="ButtonDelete"
        >
           
          <input
            id="375292223345Delete"
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr
        className="MobileClient"
      >
        <td
          className="MobileClientFam"
        >
          Петров
        </td>
        <td
          className="MobileClientIm"
        >
          Пётр
        </td>
        <td
          className="MobileClientOtch"
        >
          Петрович
        </td>
        <td
          className="MobileClientStatus"
        >
          active
        </td>
        <td
          className="MobileClientBalance"
        >
          180
        </td>
        <td
          className="ButtonEdit"
        >
           
          <input
            id="375292223346Edit"
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td
          className="ButtonDelete"
        >
           
          <input
            id="375292223346Delete"
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr
        className="MobileClient"
      >
        <td
          className="MobileClientFam"
        >
          Григорьев
        </td>
        <td
          className="MobileClientIm"
        >
          Григорий
        </td>
        <td
          className="MobileClientOtch"
        >
          Григорьевич
        </td>
        <td
          className="MobileClientStatus"
        >
          blocked
        </td>
        <td
          className="MobileClientBalance"
        >
          -220
        </td>
        <td
          className="ButtonEdit"
        >
           
          <input
            id="375292223347Edit"
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td
          className="ButtonDelete"
        >
           
          <input
            id="375292223347Delete"
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
    </tbody>
  </table>
  <div>
    <input
      onClick={[Function]}
      type="button"
      value="Добавить нового клиента"
    />
  </div>
  <div />
  <div />
</div>
`;

exports[`работа кнопки edit and delete 4`] = `
<div
  className="MobileCompany"
>
  <div
    className="CliensSorting"
  >
    <input
      onClick={[Function]}
      type="button"
      value="Все"
    />
    <input
      onClick={[Function]}
      type="button"
      value="Активные"
    />
    <input
      onClick={[Function]}
      type="button"
      value="Заблокированные"
    />
  </div>
  <table
    className="MobileCompanyClients"
  >
    <tbody
      className="Answers"
    >
      <tr>
        <td>
          фамилия
        </td>
        <td>
          имя
        </td>
        <td>
          отчество
        </td>
        <td>
          статус
        </td>
        <td>
          баланс
        </td>
        <td>
          редактировать
        </td>
        <td>
          удалить
        </td>
      </tr>
    </tbody>
    <tbody
      className="Answers"
    >
      <tr
        className="MobileClient"
      >
        <td
          className="MobileClientFam"
        />
        <td
          className="MobileClientIm"
        />
        <td
          className="MobileClientOtch"
        />
        <td
          className="MobileClientStatus"
        />
        <td
          className="MobileClientBalance"
        />
        <td
          className="ButtonEdit"
        >
           
          <input
            id="undefinedEdit"
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td
          className="ButtonDelete"
        >
           
          <input
            id="undefinedDelete"
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr
        className="MobileClient"
      >
        <td
          className="MobileClientFam"
        >
          Сидоров
        </td>
        <td
          className="MobileClientIm"
        >
          Сидор
        </td>
        <td
          className="MobileClientOtch"
        >
          Сидорович
        </td>
        <td
          className="MobileClientStatus"
        >
          active
        </td>
        <td
          className="MobileClientBalance"
        >
          250
        </td>
        <td
          className="ButtonEdit"
        >
           
          <input
            id="375292223345Edit"
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td
          className="ButtonDelete"
        >
           
          <input
            id="375292223345Delete"
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr
        className="MobileClient"
      >
        <td
          className="MobileClientFam"
        >
          Петров
        </td>
        <td
          className="MobileClientIm"
        >
          Пётр
        </td>
        <td
          className="MobileClientOtch"
        >
          Петрович
        </td>
        <td
          className="MobileClientStatus"
        >
          active
        </td>
        <td
          className="MobileClientBalance"
        >
          180
        </td>
        <td
          className="ButtonEdit"
        >
           
          <input
            id="375292223346Edit"
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td
          className="ButtonDelete"
        >
           
          <input
            id="375292223346Delete"
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
    </tbody>
  </table>
  <div>
    <input
      onClick={[Function]}
      type="button"
      value="Добавить нового клиента"
    />
  </div>
  <div />
  <div />
</div>
`;
